integration-tests/schemas/scaffolding/helpers.ts

```diff
@@ -0,0 +1,13 @@
+import { Event } from "@polkadot/types/interfaces";
+import { ISubmittableResult } from "@polkadot/types/types";
+import { pipe, map } from "rxjs";
+
+export function groupEventsByKey() {
+    return pipe(
+        map((result: ISubmittableResult) => result.events.reduce((acc, {event}) => { acc[eventKey(event)] = event.data; return acc}, {})),
+    )
+}
+
+export function eventKey(event:Event) {
+    return `${event.section}.${event.method}`;
+}
```

Comment:
Just for future reference, I used this little helper subscription in the before block to log out all the events happening while that particular test was running.

Came from https://github.com/polkadot-js/docs/blob/master/docs/api/examples/rxjs/08_system_events/index.js

```suggestion
}

export function logAllEvents(api: any, full: boolean = false) {
    api.query.system.events()    // Then we're subscribing to the emitted results
        .subscribe((events) => {
          console.log(`\nReceived ${events.length} events:`);
          // loop through the Vec<EventRecord>
          events.forEach((record) => {
          // extract the phase, event and the event types
            const { event, phase } = record;
            const types = event.typeDef;

            // show what we are busy with
            console.log(`\t${event.section}:${event.method}:: (phase=${phase.toString()})`);
            console.log(event.toHuman());

            full && console.log(`\t\t${event.meta.toHuman()}`);

            // loop through each of the parameters, displaying the type and data
            event.data.forEach((data, index) => {
                full && console.log(`\t\t\t${types[index].type}: ${data.toString()}`);
            });
          });
        });
}
```

Reply:
Not sure why we would need this?

Reply:
We don't. It is just for reference.

Reply:
So not to go into the code